src/map/battle.cpp

```diff
@@ -1839,6 +1839,13 @@ int64 battle_calc_damage(struct block_list *src,struct block_list *bl,struct Dam
 		pc_overheat(*sd, (battle_get_weapon_element(d, src, bl, skill_id, skill_lv, EQI_HAND_R, false) == ELE_FIRE ? 3 : 1));
 	}
 
+	sc = status_get_sc(bl);
```

Comment:
I'm only viewing this on the app, but was this not defined prior?

Reply:
It's assigned multiple times but the last assignment of sc is for src
`sc = status_get_sc(src);`
and we want it for bl

Reply:
Variable re-usage, the chef's kiss. Would it be better then for us to just move this check up inside the bl if-statement that ends around line 1735~? Or is it needed to be down here to be after some other check? If it's the latter, maybe add an extra comment stating it needs to be after X, Y, Z check or something so we will know for future reference!

Reply:
I'm not sure but for me putting it at the end of `battle_calc_damage` (like `damagetaken`) is a safe spot